src/Microsoft.DotNet.Darc/DarcLib/VirtualMonoRepo/VmrCloner.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.DotNet.DarcLib.Helpers;
+
+namespace Microsoft.DotNet.DarcLib.VirtualMonoRepo;
+
+/// <summary>
+/// Clones the VMR repo, and refreshes the VMR info.
+/// </summary>
+public interface IVmrCloner
+{
+    Task PrepareVmrCloneAsync(CancellationToken cancellationToken);
+}
+
+public class VmrCloner(
+    ILocalGitClient localGitClient,
+    IProcessManager processManager,
+    IVmrInfo vmrInfo,
+    string vmrLocation) : IVmrCloner
+{
+    private readonly string _vmrLocation = vmrLocation;
+
+    public async Task PrepareVmrCloneAsync(CancellationToken cancellationToken)
```

Comment:
We should not reinvent cloning git repos yet again as it's not so simple (auth, fetching, remote management, too long paths etc).
We should not have a dedicated new class.

If we need a special call for the VMR, let's just extend `RepositoryCloneManager` with a method.
But it should keep track of the VMR too potentially - so that later it can fetch it etc.

Reply:
fixed